app/controllers/api/v1/journeys_controller.rb

```diff
@@ -123,13 +124,19 @@ def update_journey_attributes
         end
       end
 
+      def timestamp
+        @timestamp ||= Time.zone.parse(data_params.dig(:attributes, :timestamp))
```

Comment:
This throws an error if the timestamp attribute is missing - might need a guard clause around that or catch the possible exception:

```ruby
Time.zone.parse(nil)
TypeError: no implicit conversion of nil into String
```

Reply:
Actually same could be true of the time zone parsing in my previous comment

Reply:
the `before_action :validate_params` takes care of ensuring `timestamp` is the approved iso8601 format (and is not nil)